x/icacallbacks/module_ibc.go

```diff
@@ -104,11 +110,13 @@ func GetTxMsgData(ctx sdk.Context, ack channeltypes.Acknowledgement, logger log.
 			logger.Error(fmt.Sprintf("cannot unmarshal ICS-27 tx message data: %s", err.Error()))
 			return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-27 tx message data: %s", err.Error())
 		}
-		return txMsgData, nil
+		// NOTE: Is this safe to assume? According to the ics-27 spec, errors on host chains should be reflected in the acknowledgement type
```

Comment:
I think we can add the following under this branch:

```
	switch len(txMsgData.Data) {
	case 0: // for SDK 0.46 and above
		return  &types.ICATxResponse{Status: types.TIMEOUT, Data: txMsgData.MsgResponses}, nil 
	default: // for SDK 0.45 and below
		return  &types.ICATxResponse{Status: types.TIMEOUT, Data: txMsgData.Data}, nil
```

Reply:
Might be misunderstanding your proposal though